### PR TITLE
Issue #585: add example for custom loss

### LINKED ISSUE #585: Add docs for custom loss using registry
## Subject of the documentation

With the introduction of the registry, using custom loss/backbone/model should be easier. The related outdated docs have been removed. New docs will be necessary before the next release.

## TODO

- [x] backbone
- [ ] loss

### PR DESCRIPTION
**Pull request by mathpluscode**
# Description

Fixes #585

## Type of change

What types of changes does your code introduce to DeepReg?

_Please check the boxes that apply after submitting the pull request._

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Documentation Update (fix or improvement on the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (if none of the other choices apply)

## Checklist

_Please check the boxes that apply after submitting the pull request._

_If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [x] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [x] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/source/docs/registry.md [hunk 1]
```diff
@@ -106,3 +108,18 @@ To register a custom backbone class, the steps are as follows
...
 for further details.
+
+### Custom Loss
+
+To register a custom image loss class, the steps are as follows
+
+1. Subclass the `tf.keras.losses.Loss` and implement a custom backbone class.
+2. Import `REGISTRY` and use the decorator `@REGISTRY.register_loss` to register the
+   custom class.
+3. Use the registered name in the config for using the registered backbone.
```
Comment (YipengHu) [inline docs/source/docs/registry.md:119]: be specific: "registered name for the new custom loss"

Comment (YipengHu) [reply docs/source/docs/registry.md:119]: and its parameter as appropriate - should we use an example with parameter? too complicated

**Review by YipengHu — CHANGES_REQUESTED** [review]

File: docs/source/docs/registry.md [hunk 2]
```diff
@@ -106,3 +108,18 @@ To register a custom backbone class, the steps are as follows
 Please check the self-contained
 [example script](https://github.com/DeepRegNet/DeepReg/blob/main/examples/custom_backbone.py)
 for further details.
+
+### Custom Loss
+
+To register a custom image loss class, the steps are as follows
```
[the remark this answers is not in the document]
Comment (mathpluscode) [reply docs/source/docs/registry.md]: fixed

File: examples/custom_image_loss.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+"""This script provides an example of using custom backbone for training."""
+import tensorflow as tf
+
+from deepreg.registry import REGISTRY
+from deepreg.train import train
+
+
+@REGISTRY.register_loss(name="custom_image_loss")
+class RootSumSquaredDifference(tf.keras.losses.Loss):
```
[the remark this answers is not in the document]
Comment (mathpluscode) [reply examples/custom_image_loss.py]: fixed

File: docs/source/docs/registry.md (hunk 1 above)
Comment (mathpluscode) [reply docs/source/docs/registry.md]: good point, for parameterized loss, I will do another separate example

**Comment by mathpluscode** [comment]
@YipengHu I've updated the docs, please have a second look ^^

**Review by YipengHu — APPROVED** [review]

**Review by kate-sann5100 — APPROVED** [review]
